### PR TITLE
Implement filtering blocks - Closes #2509

### LINKED ISSUE #2509: Use the general filter component on the latest blocks page
### Description
As a user I'd like to be able to filter the list of latest blocks by block primary properties.

### Acceptance criteria 
- Use the [general filter component](https://github.com/LiskHQ/lisk-hub/issues/2508)
according to the [designs](https://projects.invisionapp.com/d/main#/console/18539694/385494274/preview)
- I should be able to filter the blocks by
   - date (from, to)
   - delegate (username, address)
   - height (exact value)

### PR DESCRIPTION
**Pull request by slaweet**
### What issue have I solved?

Closes #2509

### How have I implemented/fixed it?

- [x] Implement the block filter dropdown component
- [x] Create useFilters hook to make the state reusable
- [x] Update the FilterBar for use on blocks page

Note that there is one edge-case bug in the API that should be fixed on the backend: https://github.com/LiskHQ/lisk-service/issues/312 

### How has this been tested?

- Go to blocks page: https://jenkins.lisk.io/test/lisk-hub/PR-2526/#/monitor/blocks
- Try filtering by various options

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)

### CODE REVIEW

File: src/components/screens/monitor/blocks/blocks.js [hunk 1]
```diff
@@ -20,76 +25,106 @@ const columnClassNames = {
...
+                      </Link>
+                    </span>
+                    <span className={columnClassNames.amount}>
+                      <LiskAmount val={block.totalAmount} />
             &nbsp;
-                    {`${t('LSK')}`}
-                  </span>
-                  <span className={columnClassNames.forged}>
-                    <LiskAmount val={block.totalForged} />
+                      {`${t('LSK')}`}
```
Comment (osvaldovega) [inline src/components/screens/monitor/blocks/blocks.js]: Hey Vit I was thinking that for avoid this fix of spacing maybe we can give some space (margin) to the LiskAmount component and then we can remove the "&nbsp".
What do you think?

File: i18n/locales/en/common.json [hunk 2]
```diff
@@ -517,6 +517,8 @@
   "[Tomorrow], hh:mm A": "[Tomorrow], hh:mm A",
   "[Yesterday], hh:mm A": "[Yesterday], hh:mm A",
   "from": "from",
+  "i.e. 10169746": "i.e. 10169746",
+  "i.e. shinekami": "i.e. shinekami",
```
Comment (yasharAyari) [inline i18n/locales/en/common.json]: `10169746` and `shinekami` are static strings that don't need to be part of the translation strings. You can use `i.e {{value}}` instead of those top strings.

File: src/components/screens/monitor/blocks/blocks.js [hunk 3]
```diff
@@ -20,76 +25,102 @@ const columnClassNames = {
...
+              <Box.Content className={styles.content}>
+                {!!blocks.data.length && (
+                <TableRow isHeader className={`${grid.row}`}>
+                  <div className={columnClassNames.id}>{t('ID')}</div>
+                  <div className={columnClassNames.height}>{t('Height')}</div>
+                  <div className={columnClassNames.date}>{t('Date')}</div>
+                  <div className={columnClassNames.transactions}>{t('Transactions')}</div>
+                  <div className={columnClassNames.generator}>{t('Generated by')}</div>
+                  <div className={columnClassNames.amount}>{t('Amount')}</div>
+                  <div className={columnClassNames.forged}>{t('Forged')}</div>
```
Comment (yasharAyari) [inline src/components/screens/monitor/blocks/blocks.js]: Please extend the `columnClassNames` and add the label of the each column to that. Then you can use a loop to generate all of these divs.

**Comment by slaweet** [comment]
All asked changes implemented. Please check again, @yasharAyari

File: src/components/screens/monitor/blocks/blocks.js [hunk 4]
```diff
@@ -2,94 +2,123 @@ import { Link } from 'react-router-dom';
...
+    className: grid['col-xs-2'],
+    getValue: block => (
+      <Link to={`${routes.accounts.pathPrefix}${routes.accounts.path}/${block.generatorAddress}`}>
+        {block.generatorUsername}
+      </Link>
+    ),
+  }, {
+    header: t('Amount'),
+    className: grid['col-xs-1'],
+    getValue: block => <LiskAmount val={block.totalAmount} token={t('LSK')} />,
```
Comment (yasharAyari) [inline src/components/screens/monitor/blocks/blocks.js]: `LSK` is the name of our token and it is the same for all languages. Please get the value of that form the token constant.

File: src/components/screens/monitor/blocks/blocks.js [hunk 5]
```diff
@@ -2,94 +2,123 @@ import { Link } from 'react-router-dom';
...
+  }, {
+    header: t('Forged'),
+    className: `${grid['col-md-1']} ${grid['col-xs-2']}`,
+    getValue: block => <LiskAmount val={block.totalForged} token={t('LSK')} />,
```
Comment (yasharAyari) [inline src/components/screens/monitor/blocks/blocks.js]: the same problem here.

File: src/components/shared/liskAmount/index.js [hunk 6]
```diff
@@ -10,11 +10,19 @@ const roundToPlaces = (value, places) => {
   return Math.round(value * x) / x;
 };
 
-const LiskAmount = ({ val, roundTo }) => (
+const LiskAmount = ({ val, roundTo, token }) => (
```
Comment (yasharAyari) [inline src/components/shared/liskAmount/index.js]: Because this component renders the amount based on different tokens, it doesn't make scence to be called `liskAmount` anymore.

File: src/components/screens/monitor/blocks/blocks.js (hunk 4 above)
Comment (slaweet) [reply src/components/screens/monitor/blocks/blocks.js]: Fixed. Please check again @yasharAyari

File: src/components/shared/liskAmount/index.js (hunk 6 above)
Comment (slaweet) [reply src/components/shared/liskAmount/index.js]: Totally agree. Should I fix it in this PR @yasharAyari ? 

There are many places that need to be updated:
```sh
$ git grep LiskAmount
src/components/screens/delegates/delegatesListView/delegateRow.js:import LiskAmount from '../../../shared/liskAmount';
src/components/screens/delegates/delegatesListView/delegateRow.js:          <LiskAmount val={rewards} />
src/components/screens/delegates/delegatesListView/delegateRow.js:          <LiskAmount val={data.vote} />
src/components/screens/explorer/transactions/transactions.js:import LiskAmount from '../../../shared/liskAmoun:...skipping...
src/components/screens/delegates/delegatesListView/delegateRow.js:import LiskAmount from '../../../shared/liskAmount';
src/components/screens/delegates/delegatesListView/delegateRow.js:          <LiskAmount val={rewards} />
src/components/screens/delegates/delegatesListView/delegateRow.js:          <LiskAmount val={data.vote} />
src/components/screens/explorer/transactions/transactions.js:import LiskAmount from '../../../shared/liskAmount';
src/components/screens/explorer/transactions/transactions.js:                            <LiskAmount val={transaction.amount} />
src/components/screens/explorer/transactions/transactions.js:                      <LiskAmount val={transaction.fee} />
src/components/screens/hwWalletLogin/selectAccount/accountCard.js:import LiskAmount from '../../../shared/liskAmount';
src/components/screens/hwWalletLogin/selectAccount/accountCard.js:          <LiskAmount val={account.balance} />
src/components/screens/monitor/blockDetails/blockDetails.js:import LiskAmount from '../../../shared/liskAmount';
src/components/screens/monitor/blockDetails/blockDetails.js:                      <LiskAmount val={blockDetails.data.reward} token={t('LSK')} />
src/components/screens/monitor/blockDetails/blockDetails.js:                      <LiskAmount val={blockDetails.data.totalFee} token={t('LSK')} />
src/components/screens/monitor/blockDetails/blockDetails.js:                      <LiskAmount val={blockDetails.data.totalForged} token={t('LSK')} />
src/components/screens/monitor/blockDetails/blockDetails.js:                      <LiskAmount val={blockDetails.data.totalAmount} token={t('LSK')} />
src/components/screens/monitor/blocks/blocks.js:import LiskAmount from '../../../shared/liskAmount';
src/components/screens/monitor/blocks/blocks.js:    getValue: block => <LiskAmount val={block.totalAmount} token="LSK" />,
src/components/screens/monitor/blocks/blocks.js:    getValue: block => <LiskAmount val={block.totalForged} token="LSK" />,
src/components/screens/wallet/transactions/amount/TransactionAmount.js:import LiskAmount from '../../../../shared/liskAmount';
src/components/screens/wallet/transactions/amount/TransactionAmount.js:              <LiskAmount val={transaction.amount} roundTo={roundTo} />
src/components/screens/wallet/transactions/transactionRow.js:import LiskAmount from '../../../shared/liskAmount';
src/components/screens/wallet/transactions/transactionRow.js:          <LiskAmount val={value.fee} token={token} />
src/components/screens/wallet/walletDetails.js:import LiskAmount from '../../shared/liskAmount';
src/components/screens/wallet/walletDetails.js:                <LiskAmount val={balance} />
src/components/shared/delegate/delegateTab.js:import LiskAmount from '../liskAmount';
src/components/shared/delegate/delegateTab.js:              <LiskAmount val={delegate.vote} token={t('LSK')} />
src/components/shared/delegate/delegateTab.js:              <LiskAmount val={delegate.rewards} />
src/components/shared/header/topBar/accountMenu/accountInfo.js:import LiskAmount from '../../../liskAmount';
src/components/shared/header/topBar/accountMenu/accountInfo.js:          <LiskAmount val={account.balance} />
src/components/shared/liskAmount/index.js:const LiskAmount = ({ val, roundTo, token }) => (
src/components/shared/liskAmount/index.js:LiskAmount.defaultProps = {
src/components/shared/liskAmount/index.js:export default LiskAmount;
src/components/shared/liskAmount/index.test.js:import LiskAmount from '.';
src/components/shared/liskAmount/index.test.js:describe('LiskAmount', () => {
src/components/shared/liskAmount/index.test.js:    const wrapper = mount(<LiskAmount val={inputValue} />, {
src/components/shared/liskAmount/index.test.js:    const wrapper = mount(<LiskAmount val={inputValue} roundTo={2} />, {
src/components/shared/searchBar/transactions.js:import LiskAmount from '../liskAmount';
src/components/shared/searchBar/transactions.js:                <LiskAmount val={transactionType.value} />
src/components/shared/votes/votesTab.js:import LiskAmount from '../liskAmount';
src/components/shared/votes/votesTab.js:                        <LiskAmount val={vote.rewards} />
src/components/shared/votes/votesTab.js:                        <LiskAmount val={vote.vote} />
src/components/shared/walletDetails/walletDetails.js:import LiskAmount from '../liskAmount';
src/components/shared/walletDetails/walletDetails.js:                  <LiskAmount val={coin.balance} />
```

Comment (slaweet) [reply src/components/shared/liskAmount/index.js]: I created a follow-up ticket for this: https://github.com/LiskHQ/lisk-hub/issues/2535
